test/CMakeLists.txt

```diff
@@ -435,3 +435,9 @@ if (NOT cxxmodules)
     )
   endif()
 endif()
+
+#--canary tests------------------------------------------------------------------------------------
+if(asserts AND NOT MSVC)
+  ROOT_EXECUTABLE(testCompilerFlags testCompilerFlags.cxx LIBRARIES Core)
```

Comment:
Can we rename this to be more specific, e.g. "checkAssertsNDEBUG"?

Reply:
Yep, that's a good idea!

Reply:
Done!